rootex/framework/entity_factory.cpp

```diff
@@ -313,3 +322,81 @@ void EntityFactory::deleteEntity(Ref<Entity> entity)
 	m_Entities.erase(entity->getID());
 	entity.reset();
 }
+
+String EntityFactory::saveEntityAsClass(Ref<Entity> entity)
```

Comment:
This function having a return value seems strange. Why would the call to EntityFactory to "save the entity hierarchy as a class on disk" return to me a String which has the path of the file created? It makes more sense to return whether the operation succeeded or not.